perception_dataset/rosbag2/rosbag2_to_non_annotated_t4_converter.py

```diff
@@ -884,9 +904,18 @@ def _generate_calibrated_sensor(
                 cam_info_topic = "/".join(topic_name_splitted[:4]) + "/camera_info"
                 info = self._bag_reader.camera_info.get(cam_info_topic)
                 if info is None:
-                    continue
-                camera_intrinsic = np.delete(info.p.reshape(3, 4), 3, 1).tolist()
-                camera_distortion = info.d.tolist()
+                    raise ValueError(f"Camera info not found for {cam_info_topic}")
+                if "image_rect" in topic_name:
+                    # image is considered as already undistorted
+                    camera_intrinsic = np.delete(info.p.reshape(3, 4), 3, 1).tolist()
+                    camera_distortion = info.d.tolist()
+                elif self._undistort_image:
+                    camera_intrinsic = np.delete(info.p.reshape(3, 4), 3, 1).tolist()
+                    camera_distortion = [0.0, 0.0, 0.0, 0.0, 0.0]
+                    camera_info = info
+                else:
+                    camera_intrinsic = info.k.reshape(3, 3).tolist()
+                    camera_distortion = info.d.tolist()
```

Comment:
Since this logic itself is a bit complicated with three conditional branch, how about isolating this as a separate function?
```suggestion
                camera_intrinsic, camera_distortion, camera_info = parse_camera_info(
                    info,
                    undistort_image=self._undistort_image,
                    is_already_rectified="image_rect" in topic_name,
                )
```

Reply:
@kminoda 
Thank you for the comment.
https://github.com/tier4/tier4_perception_dataset/pull/156/commits/35ff5f395ff43ae13591fea92d478b065460c272